hibernate-core/src/main/java/org/hibernate/boot/models/xml/internal/XmlAnnotationHelper.java

```diff
@@ -1701,4 +1682,31 @@ public static void applyCollectionClassification(
 		);
 		collectionClassification.value( classification );
 	}
+
+	public static void applyVersion(JaxbVersionImpl version, MutableClassDetails mutableClassDetails, AccessType classAccessType, XmlDocumentContext xmlDocumentContext) {
+		if ( version != null ) {
+			final AccessType accessType = coalesce( version.getAccess(), classAccessType );
+			final String versionAttributeName = version.getName();
+
+			final MutableMemberDetails memberDetails = XmlProcessingHelper.getAttributeMember(
+					versionAttributeName,
+					accessType,
+					mutableClassDetails
+			);
+			memberDetails.applyAnnotationUsage(
+					JpaAnnotations.VERSION,
+					xmlDocumentContext.getModelBuildingContext()
+			);
+			applyAttributeAccessor( version, memberDetails, xmlDocumentContext );
```

Comment:
Probably not necessary, but let's keep this aligned with `processBasicAttributes`:

```suggestion
			applyAccess( accessType, memberDetails, xmlDocumentContext );
			applyAttributeAccessor( version, memberDetails, xmlDocumentContext );
```